src/utils/ethereum.ts

```diff
@@ -1,16 +1,30 @@
 import { ethers } from "ethers";
+import { TransactionResponse, TransactionReceipt } from "@ethersproject/abstract-provider";
 import hermezjs from "@hermeznetwork/hermezjs";
+import { BigNumber } from "ethers";
 
-import { ETHER_TOKEN_ID, DEPOSIT_TX_TIMEOUT } from "../constants";
+import { ETHER_TOKEN_ID, DEPOSIT_TX_TIMEOUT } from "src/constants";
+
+// domain
+import { Wallet, Token, ISOStringDate } from "src/domain/hermez";
 
 /**
  * Fetches token balances in the user's Ethereum account. Only for those tokens registered in Hermez and Ether.
  * Throws an error if the user has no balances for any registered token in Hermez or an error comes up from fetching the balances on-chain.
  *
- * @param {Object[]} hermezTokens - List of registered tokens in Hermez
+ * @param {Token[]} hermezTokens - List of registered tokens in Hermez
  * @returns {Promise} - Array of { balance, token } where balance is a Number and token is the Token schema returned from the API.
  */
-async function getTokens(wallet, hermezTokens) {
+function getTokens(
+  wallet: Wallet,
+  hermezTokens: Token[]
+): Promise<
+  {
+    balance: BigNumber;
+    token: Token;
+  }[]
+> {
+  // ToDo: Can we remove this check below?
```

Comment:
If you receive a `Wallet` as a parameter, then we can remove this check yep.

Reply:
Nice! Done